config/jest/index.js

```diff
@@ -24,3 +26,5 @@ const toHaveNoDAPViolations = require('./matchers/toHaveNoDAPViolations');
 // For more information, check out the docs here:
 // https://jestjs.io/docs/en/configuration.html#setupfilesafterenv-array
 expect.extend({ toHaveNoAxeViolations, toHaveNoDAPViolations });
+
+global.ResizeObserver = resizeObserverMock;
```

Comment:
Instead of adding this as a global object for all tests, I would just mock it for the gradient tests:


```js
const observeMock = jest.fn();
const disconnectMock = jest.fn();

beforeAll(() => {  
    window.ResizeObserver = jest.fn(function ResizeObserver() {
      this.observe = observeMock;
      this.disconnect = disconnectMock;
  });
});

beforeEach(() => {
  observeMock.restoreMock();
  disconnectMock.restoreMock();
});

afterAll(() => {
  window.ResizeObserver = undefined;
});
```

By doing this, you don't need to dig deep into the structure of a mock function that relies on the order of tests.

Reply:
Thanks! I much prefer this

Reply:
I had some challenges integrating this inline (as indicated as a comment in the code), but I've used your suggestion as a foundation for how this is structured - let me know your thoughts!

Reply:
Yeah, I think that the changes you made improve the readability and maintainability of the tests 👍